src/solver/quasinewton/lbfgs.rs

```diff
@@ -46,8 +50,22 @@ impl<L, P> LBFGS<L, P> {
             m,
             s: VecDeque::with_capacity(m),
             y: VecDeque::with_capacity(m),
+            tol_precision: std::f64::EPSILON.sqrt(),
+            tol_cost: std::f64::EPSILON,
         }
     }
+
+    /// Sets tolerance for precision stopping criterion
+    pub fn with_tol_precision(mut self, tol_precision: f64) -> Self {
```

Comment:
I think it would help to be more explicit that this tolerance is applied on the gradient. E.g. in [scipy they have a pgtol](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.fmin_l_bfgs_b.html) for projected gradient tolerance,
>  The iteration will stop when max{|proj g_i | i = 1, ..., n} <= pgtol where pg_i is the i-th component of the projected gradient.

How about `with_tol_grad`? I realize this makes it a bit more difficult to have common API for tolerance across solvers. If you want one main tolerance, maybe `with_tol` and make it very explicit in the docsting that it's taken on the gradient? I just find the word combination "tolerance precision" confusing.

Reply:
Good point, thanks. I've opted for `with_tol_grad`. I've come to think that trying to have a common API across solvers may be counter-productive as tolerances often mean slightly different things for different solvers.